From 28b668883fff4e8b1b75bac5e1590447f783c0c7 Mon Sep 17 00:00:00 2001
From: Robert Yang <liezhi.yang@windriver.com>
Date: Sat, 24 May 2014 14:39:56 +0800
Subject: [PATCH] devices.py: don't check partitions on /LiveOS/rootfs.img

Don't check the partitions on the /LiveOS/rootfs.img since it would cost
too much memory when work on aufs (ISO), and the rootfs.img doens't
contain any partition info, so this should be harmless.

Upstream-Status: Pending

Signed-off-by: Robert Yang <liezhi.yang@windriver.com>
---
 pyanaconda/storage/devices.py | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/pyanaconda/storage/devices.py b/pyanaconda/storage/devices.py
index a52e464..5ddf11c 100644
--- a/pyanaconda/storage/devices.py
+++ b/pyanaconda/storage/devices.py
@@ -556,7 +556,12 @@ class StorageDevice(Device):
 
     @property
     def partedDevice(self):
-        if self.exists and self.status and not self._partedDevice:
+        # Don't check the partitions on the /LiveOS/rootfs.img since it
+        # would cost too much memory when work on aufs (ISO), and the
+        # rootfs.img doens't contain any partition info, so this is
+        # harmless.
+        if self.exists and self.status and not self._partedDevice \
+                and self.path != "/LiveOS/rootfs.img":
             log.debug("looking up parted Device: %s" % self.path)
 
             # We aren't guaranteed to be able to get a device.  In
-- 
1.8.3.1

